arch/arm/core/aarch32/Kconfig

```diff
@@ -126,6 +126,13 @@ config COMPILER_ISA_THUMB2
 config NUM_IRQS
 	int
 
+config SHARED_ISR_TABLE
+	bool "Allow IRQS to share multiple ISRS"
+	default y
+	help
+	  Some ARM peripherals share an IRQ and this option enables the system
+	  to call all ISRs attached to a pending IRQ.
```

Comment:
1. Why is this AArch32 specific?
2. This is confusing for someone that is used to the Zephyr nomenclature. The ISR table is something different than the vector table, yet this symbol is used to modify both.

Reply:
> 1. Why is this AArch32 specific?

Agreed. In case of GD32VF103 (RISC-V) ADC0/1 also share the same IRQ line. I'd expect this option to be cross-arch (not necessary to provide all implementations, if arch specific quirks are needed)

Reply:
BTW, this is usually why (as it was suggested before) you want to have an `ARCH_HAS_*` symbol together with the actual symbol enabling the feature.

Reply:
> 1. Why is this AArch32 specific?

I only tested this PR on AArch32 boards and I thought the feature would be made available to other architectures by someone that could make sure it worked on said architecture.

> 2. This is confusing for someone that is used to the Zephyr nomenclature. The ISR table is something different than the vector table, yet this symbol is used to modify both.

I'll change the option name to ARCH_SUPPORT_SHARED_IRQ

Reply:
> I only tested this PR on AArch32 boards and I thought the feature would be made available to other architectures by someone that could make sure it worked on said architecture.

We have CI for that. And a bunch of QEMU boards as well.

Reply:
If all looks well with the AArch32 changes, I'll add the other architectures.